src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -579,6 +585,26 @@ private static ExpressionSyntax SkipParensAndNullSuppressions(ExpressionSyntax e
             }
             else
             {
+                if (expression.Type?.SpecialType == SpecialType.System_String && inputType.IsSpanOrReadOnlySpanChar())
+                {
+                    if (MessageID.IDS_FeatureSpanCharConstantPattern.CheckFeatureAvailability(diagnostics, Compilation, node.Location))
+                    {
+                        // report missing member and use site diagnostics
+                        if (inputType.IsReadOnlySpanChar())
```

Comment:
>inputType.IsReadOnlySpanChar() [](http://example.com/codeflow?start=28&length=30)

Minor: Consider using conditional expression in `GetWellKnownTypeMember()` call rather than `if { } else { }`.